osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DeviceManagementServiceTest.java

```diff
@@ -0,0 +1,217 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.opensmartgridplatform.adapter.domain.core.application.mapping.DomainCoreMapper;
+import org.opensmartgridplatform.adapter.domain.core.infra.jms.core.OsgpCoreRequestMessageSender;
+import org.opensmartgridplatform.adapter.domain.core.infra.jms.ws.WebServiceResponseMessageSender;
+import org.opensmartgridplatform.domain.core.entities.Device;
+import org.opensmartgridplatform.domain.core.entities.Organisation;
+import org.opensmartgridplatform.domain.core.exceptions.UnknownEntityException;
+import org.opensmartgridplatform.domain.core.services.DeviceDomainService;
+import org.opensmartgridplatform.domain.core.services.OrganisationDomainService;
+import org.opensmartgridplatform.domain.core.valueobjects.CdmaSettings;
+import org.opensmartgridplatform.domain.core.valueobjects.Certification;
+import org.opensmartgridplatform.domain.core.valueobjects.DeviceLifecycleStatus;
+import org.opensmartgridplatform.domain.core.valueobjects.EventNotificationType;
+import org.opensmartgridplatform.dto.valueobjects.EventNotificationMessageDataContainerDto;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalException;
+import org.opensmartgridplatform.shared.infra.jms.RequestMessage;
+import org.opensmartgridplatform.shared.infra.jms.ResponseMessage;
+import org.opensmartgridplatform.shared.infra.jms.ResponseMessageResultType;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class DeviceManagementServiceTest {
+	
+	@Mock
+	private TransactionalDeviceService transactionalDeviceService;
+	
+	@Mock
+	private DeviceDomainService deviceDomainService;
+	
+	@Mock
+	private DomainCoreMapper domainCoreMapper;
+	
+	@Mock
+	private OsgpCoreRequestMessageSender osgpCoreRequestManager;
+	
+	@Mock
+	private WebServiceResponseMessageSender webServiceResponseMessageSender;
+
+	@Mock
+	private OrganisationDomainService organisationDomainService;
+	
+	@InjectMocks
+	private DeviceManagementService deviceManagementService;
+
+	@Captor
+	private ArgumentCaptor<RequestMessage> argumentReqM;
+
+	@Captor
+	private ArgumentCaptor<ResponseMessage> argumentResM;
+
+	@Captor
+	private ArgumentCaptor<String> argumentStringOne;
```

Comment:
give this field a descriptive name like argumentMessageType

Reply:
Renamed this field and added a new argumentDeviceIdentification field, since argumentStringOne was used for capturing both messageType and deviceIdentification.